apps/api/app/agents/middleware/vfs_summarization.py

```
@@ -0,0 +1,229 @@
+        self, state: AgentState[Any], runtime: Runtime[Any]
+    ) -> dict[str, Any] | None:
+        """
+        Called before each model invocation.
+
+        Archives to VFS if summarization will occur, then delegates to parent.
+        """
+        # Check if we should archive (if parent will summarize)
+        if self.vfs_enabled and self._should_trigger_summarization(state):
+            try:
+                await self._archive_to_vfs(state, runtime)
+            except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)